src/init/application/application.ts

```diff
@@ -15,90 +15,93 @@ import * as chalk from 'chalk';
 import * as path from 'path';
 import Generator = require('yeoman-generator');
 
-export class ApplicationGenerator extends Generator {
-  props: any;
+export function createApplicationGenerator(templateName: string):
```

Comment:
doc that the template name is both the display name and where the template files are loaded from